sklearn/metrics/_classification.py

```diff
@@ -1911,10 +1956,10 @@ def precision_recall_fscore_support(
         weights = None
 
     if average is not None:
-        assert average != "binary" or len(precision) == 1
-        precision = _nanaverage(precision, weights=weights)
-        recall = _nanaverage(recall, weights=weights)
-        f_score = _nanaverage(f_score, weights=weights)
+        assert average != "binary" or precision.shape[0] == 1
+        precision = float(_nanaverage(precision, weights=weights))
+        recall = float(_nanaverage(recall, weights=weights))
+        f_score = float(_nanaverage(f_score, weights=weights))
```

Comment:
so we're also converting these results to python types at the same time, a link to https://github.com/scikit-learn/scikit-learn/issues/27339 would be nice in the description or the PR probably.

Reply:
Added in the description of the PR